### PR TITLE
LG-394 Remove old proofing logic

### PR DESCRIPTION
**Pull request by jmhooper**
**Why**: We no longer use the old logic and it is liable to confuse
people (ref: https://github.com/18F/identity-proofer-gem/pull/42)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,56 +6,44 @@
...
-confirmation.success? # => true
-confirmation.errors # A hash of errors if errors occur
-
-# Submit financials w/ a bank account
-confirmation = agent.submit_financials(
-  bank_account: '123456789',
-  bank_account_type: :savings,
-  bank_routing: '987654321',
+# Create a proofer subclass:
+class FooProofer
```
Comment (davemcorwin) [inline README.md]: need to subclass `Proofer::Base` :)

File: README.md [hunk 2]
```diff
@@ -6,56 +6,44 @@
...
+
+  stage :resolution
+
+  proof :foo_proof # this also takes a block
+
+  def foo_proof(applicant, result)
+    resolution = get_resolution # resolve the identity
+
+    # if something isn't verified, add an error
+    resolution.add_error(:first_name, 'Does not match')
```
Comment (davemcorwin) [inline README.md]: should be `result.add_error`

File: README.md [hunk 3]
```diff
@@ -6,56 +6,44 @@
...
+
+    # if something goes wrong, raise an error
+    raise 'failed to auth with the proofing vendor'
+  end
+end
+
+# Use a vendor subclass
+proofer = FooProofer.new.proof(
```
Comment (davemcorwin) [inline README.md]: The thing returned here is a `Result` not the proofer :), so might want to name the variable something more semantic. Also below, you call it `subject`.

**Review by davemcorwin — COMMENTED** [review]
Sawheet! Love gettin rid of the cruft!

Looks like we need to include `rake` as a dependency...

The example in the readme is great, it may be nice at some point to put in a full example with context, and whatnot.

We could also take this opportunity to upgrade to ruby 2.5 like the idp and update `Proofer::Result.errors` to use `transform_values`.

**Comment by jmhooper** [comment]
@davemcorwin: I cleaned up the README and upgrade ruby to 2.5.1. Lmk what you think.

**Review by davemcorwin — APPROVED** [review]
LGTM

do we have to do something to add it to snyk?

File: proofer.gemspec [hunk 4]
```diff
@@ -27,7 +27,6 @@ Gem::Specification.new do |s|
   s.required_ruby_version = '>= 2.3.0'
```
Comment (davemcorwin) [inline proofer.gemspec]: Forgot to mention you should prob upgrade the ruby dependency here as well

**Comment by jmhooper** [comment]
I setup the snyk webhook so hopefully it'll show up down the line.

File: proofer.gemspec (hunk 4 above)
Comment (jmhooper) [reply proofer.gemspec]: good catch, I'll fix

**Pull request merged**